urld73c47f3e5_lindenb_dataindexer_tgz-pJ8-test_github_lindenb_dataindexer_SimpleIntegerJ8/src/com/github/lindenb/dataindexer/SecondaryDataWriter.java

```diff
@@ -0,0 +1,298 @@
+				 {
+				 ObjectAndOffset<K> oao=rootFile.read();
+				 buffer.add(oao);
+				 }
+			 LOG.info("sorting buffer N="+buffer.size());
+			// Sort M elements
+			 Collections.sort(buffer,this.objectAndOffsetComparator);
+			 if(prevFile==null)
+			 	{
+				 prevFile=new FileAndSize();
+				 prevFile.file=tmpFile();
+				 LOG.info("saving buffer to="+prevFile);
+				 prevFile.openWrite();
+				 for (ObjectAndOffset<K> oao:buffer)
+				 	{
+					prevFile.write(oao);
+				 	}
+				prevFile.close();
+			 	}
+			 else
+			 	{
+				FileAndSize nextFile=new FileAndSize();
+				nextFile.file=tmpFile();
+				nextFile.openWrite();
+				
+				ObjectAndOffset<K> diskItem=null;
+				ObjectAndOffset<K> objectItem=null;
+				Iterator<ObjectAndOffset<K>> iter=buffer.iterator();
+				
+				prevFile.openRead();
+				
+				LOG.info("merging "+prevFile);
+				
+				for(;;)
+					{
+					if(objectItem==null && !iter.hasNext()) break;
+					if(diskItem==null && prevFile.count<=0) break;
+					if(objectItem==null)
+						{
+						objectItem=iter.next();
+						}
+					if(diskItem==null)
+						{
+						diskItem=prevFile.read();
+						}
+					if(this.objectAndOffsetComparator.compare(objectItem, diskItem)<=0)
+						{
+						nextFile.write(objectItem);
+						objectItem=null;
+						}
+					else
+						{
+						nextFile.write(diskItem);
+						diskItem=null;
+						}
+					}
+				if(objectItem!=null)
+					{
+					if(diskItem!=null) throw new IllegalStateException();
+					nextFile.write(objectItem);
+					}
+				while(iter.hasNext())
+					{
+					objectItem=iter.next();
+					nextFile.write(objectItem);
+					}
+				if(diskItem!=null)
+					{
+					if(objectItem!=null) throw new IllegalStateException();
+					nextFile.write(diskItem);
+					}
+				while(prevFile.count>0)
+					{
+					diskItem=prevFile.read();
+					nextFile.write(diskItem);
+					}
+					
+				prevFile.close();
+				nextFile.close();
+				prevFile.file.delete();
+				prevFile=nextFile;
+			 	}
+			 }
+		rootFile.close();
+		
+		
+			 }
+		 LOG.info("saving result");
+		RandomAccessOutput dataOut=getConfig().getRandomAccessFactory().openForWriting(
+				getConfig().getDataFile()
+				);
+		RandomAccessFile indexFile=null;
+		if(!getConfig().isFixedSizeof())
+			{
+			indexFile=new RandomAccessFile(getConfig().getIndexFile(), "rw");
+			}
+		
+		if(prevFile!=null)//size()>0
+			{
+			if(prevFile.count!=this.numberOfItems) throw new IOException();	
+			prevFile.openRead();
+
+			while(prevFile.count>0)
+				{
+				ObjectAndOffset<K> o=prevFile.read();
+				long offset=-1L;
+				
+				
+				if(indexFile!=null)
+					{
+					offset=dataOut.getOffset();
+					indexFile.writeLong(offset);
+					}
+				DataOutputStream daos=new DataOutputStream(dataOut);
+				getDataBinding().writeObject(o,daos);
+				daos.flush();
+				}
+			}
+		
+		if(indexFile!=null)
+			{
+			indexFile.close();
+			}
+		
+		dataOut.flush();
+		dataOut.close();
+		
+		prevFile.close();
+		prevFile.file.delete();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of deleting a file on this line should be checked and failure should be logged. java.io.File.delete() will simply return false if it fails to delete the file, for example, when the file does not exist. It will not throw unless there is a security exception.